src/modules/keyboardmanager/dll/dllmain.cpp

```diff
@@ -267,13 +267,26 @@ class KeyboardManager : public PowertoyModuleIface
     virtual void enable()
     {
         m_enabled = true;
+        std::unique_lock<std::mutex> keyRemapLock(keyboardManagerState.singleKeyReMap_mutex);
+        bool atleastOneKeyRemap = (keyboardManagerState.singleKeyReMap.size() > 0);
+        keyRemapLock.unlock();
+        std::unique_lock<std::mutex> shortcutRemapLock(keyboardManagerState.osLevelShortcutReMap_mutex);
+        bool atleastOneShortcutRemap = (keyboardManagerState.osLevelShortcutReMap.size() > 0);
+        shortcutRemapLock.unlock();
+        // Log telemetry
+        Trace::EnableKeyboardManager(true);

```

Comment:
@saahmedm @arjunbalgovind.  It seems odd to me to do the 'atleastone' logic on the client.  Woudn't this be better just to report the remap.size and you could do the filtering while generating a report?  What if you wanted to know how many shortcuts, or keys get remapped on average?  My preference for telemetry, is that the client should fire the events as generic as possible, and the telemetry service should do the filtering / aggregation logic if possible.  It tends to be more flexible that way.  This way if your questions change you still have the data.

Reply:
I agree with that. I was not sure if the telemetry service has the options for combining two different metrics together. There are already two separate metrics for the remap sizes and a separate metric for if KBM is enabled, so it should be possible to do this "atleast one" logic at the report end.